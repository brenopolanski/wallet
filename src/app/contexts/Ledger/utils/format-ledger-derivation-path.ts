import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
