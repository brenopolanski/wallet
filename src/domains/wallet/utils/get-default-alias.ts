import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
