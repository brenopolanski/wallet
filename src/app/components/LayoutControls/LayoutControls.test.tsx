import React from "react";
import { fireEvent, render } from "utils/testing-library";

import { LayoutControls } from "./LayoutControls";

describe("LayoutControls", () => {
	it("should render", () => {
		const { asFragment, getByTestId } = render(<LayoutControls />);

		expect(getByTestId("LayoutControls__grid")).toBeInTheDocument();
		expect(getByTestId("LayoutControls__list")).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it.each([
		["onSelectGridView", "LayoutControls__grid--icon"],
		["onSelectListView", "LayoutControls__list--icon"],
	])("should call %s callback if provided", (callback, element) => {
		const function_ = jest.fn();

		const { getByTestId } = render(<LayoutControls {...{ [callback]: function_ }} />);

		fireEvent.click(getByTestId(element));

		expect(function_).toHaveBeenCalledWith(expect.objectContaining({ nativeEvent: expect.any(MouseEvent) }));
	});

	it.each([
		["onSelectGridView", "LayoutControls__grid--icon"],
		["onSelectListView", "LayoutControls__list--icon"],
	])("should not call %s callback if not provided", (callback, element) => {
		const function_ = jest.fn();

		const { getByTestId } = render(<LayoutControls />);

		fireEvent.click(getByTestId(element));

		expect(function_).not.toHaveBeenCalled();
	});
});
