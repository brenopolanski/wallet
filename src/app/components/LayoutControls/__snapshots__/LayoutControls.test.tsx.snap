// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutControls should render 1`] = `
<DocumentFragment>
  <div
    class="flex items-center space-x-1"
  >
    <div
      data-testid="LayoutControls__list"
    >
      <div
        class="group"
      >
        <button
          class="ControlButton__ControlButtonStyled-sc-1l8cuc8-0 ckIXFQ ring-focus"
          data-testid="LayoutControls__list--icon"
        >
          <div
            class="sc-bdvvtL qGXhc"
            height="20"
            width="20"
          >
            <svg>
              list.svg
            </svg>
          </div>
        </button>
      </div>
    </div>
    <div
      data-testid="LayoutControls__grid"
    >
      <div
        class="group"
      >
        <button
          class="ControlButton__ControlButtonStyled-sc-1l8cuc8-0 ckIXFQ ring-focus active"
          data-testid="LayoutControls__grid--icon"
        >
          <div
            class="sc-bdvvtL qGXhc"
            height="20"
            width="20"
          >
            <svg>
              grid.svg
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
